_targets_eval_postprocessing.R

```diff
@@ -301,6 +301,15 @@ head_to_head_targets <- list(
   # Get the full set of quantiles, filtered down to only states and
   # forecast dates with sufficient wastewater for both ww model and hosp only
   # model. Then join the convergence df
+  tar_target(
+    name = max_nowcast_days,
+    command = as.numeric(
+      max(all_ww_hosp_quantiles$forecast_date) -
+        max(all_ww_hosp_quantiles$date[!is.na(
+          all_ww_hosp_quantiles$calib_data
+        )])
```

Comment:
I'm not convinced this is the quantity you want to be computing. Can you explain how you're trying to set the `max_nowcast_days` here? As the set of days before the forecast date without _any_ admissions data on the date the forecast was made? Do you want it a different value for each `forecast_date`/`scenario`/`location` trio?

Reply:
Here I am just finding a single value which represents the number of days between the forecast date and the last day of observed data. I am just using the maximum value to find this so I can get a single number. In practice, this is the same for all `forecast_date`/`scenario`/`locations`, data ends on a Friday and the forecast is made the following Monday. 

In theory, this could be different for each `forecast_date`/`scenario`/`location` , which of course means you couldn't just pass it as a function argument but would need to calculate it based on the `calib_data` within the `df` (which is what I was originally doing with the `add_horizons_to_quantiles()` function.

I am not following how there would be a way to have `max_nowcast_days` passed as an argument to `add_horizons()` without passing a single scalar and assuming it is the same for all combinations. This is why I broke the functions up originally, because with the quantiles, we have `calib_data` and with the scores, I have already filtered to remove days with observed data (we don't score those days).

I probably made this more complicated than it needed to be!

Reply:
@dylanhmorris lmk what you think, when this is done I'll need to port the changes to my fig3 branch

Reply:
If `max_nowcast_days` is a vector of the same length as the dataframe, the comparisons will work, as `case_when` is vectorized.

That said, if your goal is to start the "nowcast" on the period on the first day without observed hospitalizations, the simplest thing to do is to have a "`last_hosp_data_date` column in both all dataframes that get passed to `add_horizons()` and compare directly to that. Is that easy to do? If so, we should do it here. If not, we can stick with this approach.

Reply:
I think adding that column to both dataframes is the most elegant solution! I can just add a target that creates that mapping of `location`/`forecast_date`/`last_hosp_data_date` and then left join it to both dataframes

Reply:
Sounds good. Ping me when ready for review.  Minor note: since every entry in the `score_df` and `quantile_df` should have an entry in the `last_data_date_df` (not committed to these names, just for clarity), I would do `right_join(last_data_date_df, score_df, unmatched = "error")` or `left_join(score_df, last_data_date_df, unmatched = "error")`. See https://dplyr.tidyverse.org/reference/mutate-joins.html

Reply:
`last_data_date_df` will contain a larger subset of combinations of `location` and `forecast_date` than `scores_filtered` and `hosp_quantiles_filtered`, bc both `score_df` and `quantile_df` will have been filtered to only include locations and forecast dates where the wastewater is sufficient and the model has converged. 

https://github.com/CDCgov/wastewater-informed-covid-forecasting/blob/b0d50364e996b8ffedf2f7f1e7797efa1b56e8df/_targets_eval_postprocessing.R#L354

So if we flag it to error if unmatched, the "extra" mappings will cause an error.

Reply:
Apologies, I was wrong about the behavior of that argument. In that case, just make sure to do a join that removes the non-matches, i.e.  either `right_join(last_data_date_df, score_df)` or `left_join(score_df, last_data_date_df)